storage/spider/spd_db_mysql.cc

```diff
@@ -4181,18 +4181,108 @@ int spider_db_mbase_util::open_item_func(
+        if (!strncasecmp("rand", func_name, func_name_length) &&
 #ifdef SPIDER_Item_args_arg_count_IS_PROTECTED
-        !item_func->argument_count()
+           !item_func->argument_count()
 #else
-        !item_func->arg_count
+           !item_func->arg_count
 #endif
-      ) {
-        if (str)
-          str->length(str->length() - SPIDER_SQL_OPEN_PAREN_LEN);
-        DBUG_RETURN(spider_db_open_item_int(item_func, NULL, spider, str,
-          alias, alias_length, dbton_id, use_fields, fields));
+        ) {
+          if (str)
+            str->length(str->length() - SPIDER_SQL_OPEN_PAREN_LEN);
+          DBUG_RETURN(spider_db_open_item_int(item_func, NULL, spider, str,
+            alias, alias_length, dbton_id, use_fields, fields));
+        } else if (!strncasecmp("trim", func_name, func_name_length) &&
+          item_count == 2) {
+          /* item_count == 1 means this TRIM() is without a remove_str */
+          item = item_list[0];
+          Item *item_tmp = item_list[1];
+          if (str) {
+            if (item_tmp->type() == Item::STRING_ITEM) {
+              /* 1. append 'TRIM(BOTH ' */
+              if (str->reserve(SPIDER_SQL_TRIM_LEN + SPIDER_SQL_OPEN_PAREN_LEN +
+                               SPIDER_SQL_TRIM_BOTH_LEN))
+                DBUG_RETURN(HA_ERR_OUT_OF_MEM);
+              str->q_append(SPIDER_SQL_TRIM_STR, SPIDER_SQL_TRIM_LEN);
+              str->q_append(SPIDER_SQL_OPEN_PAREN_STR,
+                            SPIDER_SQL_OPEN_PAREN_LEN);
+              str->q_append(SPIDER_SQL_TRIM_BOTH_STR, SPIDER_SQL_TRIM_BOTH_LEN);
+              /* 2. append "remove_str"*/
+              if ((error_num = spider_db_print_item_type(
+                      item_tmp, NULL, spider, str, alias, alias_length, dbton_id,
+                      use_fields, fields)))
+                DBUG_RETURN(error_num);
+              /* 3. append ' FROM ' */
+              if (str->reserve(SPIDER_SQL_FROM_LEN))
+                DBUG_RETURN(HA_ERR_OUT_OF_MEM);
+              str->q_append(SPIDER_SQL_FROM_STR, SPIDER_SQL_FROM_LEN);
+              /* 4. append `field` */
+              if ((error_num = spider_db_print_item_type(
+                      item, NULL, spider, str, alias, alias_length, dbton_id,
+                      use_fields, fields)))
+                DBUG_RETURN(error_num);
+              /* 5. append ')' */
+              if (str->reserve(SPIDER_SQL_CLOSE_PAREN_LEN))
+                DBUG_RETURN(HA_ERR_OUT_OF_MEM);
+              str->q_append(SPIDER_SQL_CLOSE_PAREN_STR,
+                            SPIDER_SQL_CLOSE_PAREN_LEN);
+            }
+          }
+          item_count -= 2;
+          break;
+        }
+            } else if (func_name_length == 5) {
+        if ((!strncasecmp("ltrim", func_name, func_name_length) ||
+            !strncasecmp("rtrim", func_name, func_name_length)) &&
+            (item_count == 2)) {
+                    /* item_count = 2 means this LTRIM/RTRIM is actually TRIM(LEADING/TRAILING) */
```

Comment:
`TRIM(LEADING ...)` and `TRIM(TRAILING ...)` seems to be handled here. This seems not to be easy to understand at first glance. So, could you add a little more explanation on why these are handled here?

Reply:
The `func_name` here is `LTRIM` for `TRIM(LEADING ...)`, and `RTRIM` for `TRIM(TRAILING ...)`, rather than `TRIM`, thus they can't be handled in the
```cpp
if (func_name_length == 4 && !strncasecmp("trim", func_name, func_name_length)) {
```
branch.

Reply:
@nayuta-yanagisawa Do you have any better ideas to improve handling the trim leading and trim trailing functions?

Reply:
The implementation looks OK to me. I just wanted you to add a comment explaining that `func_name` of `TRIM(LEADING ...)` is `ltrim` and that of `TRI(TRAILING ...)` is `rtrim`.

> The func_name here is LTRIM for TRIM(LEADING ...), and RTRIM for TRIM(TRAILING ...), rather than TRIM, thus they can't be handled in the